src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -405,13 +417,30 @@ const mesh::PtrData &ParticipantConfiguration::getData(
     const mesh::PtrMesh &mesh,
     const std::string &  nameData) const
 {
+  // if getData is called for global data, then the config file must incorrectly have a mesh name with the global data.
+  PRECICE_CHECK(!(_meshConfig->getDataConfiguration()->hasGlobalDataName(nameData)),
+                "Participant \"{}\" specifies mesh name \"{}\" for global data \"{}\", but global data are not associated with meshes. "
+                "Please remove mesh=\"{}\".",
```

Comment:
Yes, this would appear in the case when the xml specifies a mesh name with a global data:
```xml
    <participant name="SolverOne">
      <write-data name="GlobalData1" mesh="MeshOne"/>
    </participant>
```
So we want to suggest the user to remove `mesh="MeshOne"`

Reply:
Ah, ok. Maybe we could change to "Please remove the attribute mesh=... from the corresponding write-data tag".

Reply:
done